backend/src/index.ts

```diff
@@ -402,6 +433,186 @@ app.get("/societies", async (req, res: Response) => {
   return res.status(200).json(societies);
 });
 
+//Lets a user join a society
+app.post("/user/society/join",
+  async (req: TypedRequest<societyIdBody>, res: Response) => {
+    //get userid from session
+    const sessionFromDB = await validateSession(
+      req.session ? req.session : null
+    );
+    if (!sessionFromDB) {
+      return res.status(401).json({ message: "Invalid session provided." });
+    }
+
+    const userID = sessionFromDB.userId;
+
+    //Make sure a society actually exists
+    const societyId = await prisma.society.findFirst({
+      where: {
+        id: req.body.societyId
+      },
+      select: {
+        id: true
+      }
+    })
+
+    if (!societyId) {
+      return res.status(400).json({message: "Invalid society"})
+    }
+
+    //Connect society and user
+    const result = await prisma.society.update({
+      where: {
+        id: req.body.societyId,
+      },
+      data: {
+        members: {
+          connect: {
+            id: userID,
+          },
+        },
+      },
+    });
+
+    return res.status(200).json({message: "ok"});
+  }
+);
+
+app.post("/user/society/leave", async (req: TypedRequest<societyIdBody>, res: Response) => {

```

Comment:
consider `app.delete('/user/society')`

Reply:
I'll also do the same for user/event/unattend.